addons/mass_mailing_sms/models/mailing_mailing.py

```diff
@@ -367,3 +369,36 @@ def _get_default_ab_testing_campaign_values(self, values=None):
                 campaign_values['name'] = _("A/B Test: %s", sms_subject)
             campaign_values['ab_testing_sms_winner_selection'] = self.ab_testing_sms_winner_selection
         return campaign_values
+
+    def get_chars_counts(self):
+        """Get character counts for sms widget computation of characters and sms counts.
+
+        :return: Character counts used for links, formatted as `{optOutLink: int, link: int}`.
+        """
+        # opt-out is formatted as "\nSTOP SMS: [base_url]/sms/[mailing_id]/[trace_code]"
+        # links are formatted as "[base_url]/r/[link_tracker_code]/s/[sms_id]"
+        # Note: `url_join` will add a '/' as necessary.
+        if len(self) > 1:
+            return UserError("Call on zero or one record.")
```

Comment:
Actually we don't really need to raise a UserError with a proper message, this is either called by JS, either people did custom code, but we won't be sending the error to people in our flows. A "self.ensure_one() if self" is probably sufficient ?